tests/testthat/test-shinytest2-adtteSpec.R

```diff
@@ -0,0 +1,226 @@
+  expect_error(h_km_mae_to_adtte(
+    good_adtte,
+    mae,
+    genes = hermes::gene_spec("GeneID:1820"),
+    experiment_name = 1
+  ))
+  expect_error(h_km_mae_to_adtte(
+    good_adtte,
+    mae,
+    genes = 1520
+  ))
+})
+
+test_that("h_km_mae_to_adtte warns when patients are not in ADTTE and therefore removed", {
+  mae <- hermes::multi_assay_experiment
+  adtte <- scda::synthetic_cdisc_data("rcd_2021_07_07")$adtte %>%
+    dplyr::mutate(CNSR = as.logical(.data$CNSR))
+
+  # Example with no matched patient IDs.
+  bad_adtte <- adtte %>% dplyr::mutate(USUBJID = paste0("bla-", USUBJID))
+  result <- expect_warning(
+    h_km_mae_to_adtte(
+      bad_adtte,
+      mae,
+      genes = hermes::gene_spec("GeneID:1820"),
+      experiment_name = "hd2"
+    ),
+    "from gene data set were lost"
+  )
+})
+
+test_that("h_km_mae_to_adtte fails as expected if USUBJID in MAE colData is different from sample map", {
+  mae <- MultiAssayExperiment::MultiAssayExperiment(
+    experiments = list(
+      a = hermes::HermesDataFromMatrix(
+        matrix(1:4, 2, 2, dimnames = list(c("ENSG1", "ENSG2"), c("A", "B"))),
+        colData = data.frame(USUBJID = c("A", "B")) # USUBJID on experiment.
+      )
+    ),
+    colData = data.frame(
+      USUBJID = c("C", "D"), # USUBJID on MAE.
+      row.names = c("A", "B")
+    )
+  )
+  adtte <- data.frame(USUBJID = c("A", "B"))
+
+  expect_error(
+    h_km_mae_to_adtte(
+      adtte,
+      mae,
+      genes = hermes::gene_spec("ENSG1"),
+      experiment_name = "a"
+    ),
+    "Must be a subset of {'C','D'}, but has additional elements {'A','B'}",
+    fixed = TRUE
+  )
+})
+
+test_that("h_km_mae_to_adtte fails as expected if USUBJID in experiment colData is different from sample map", {
+  mae <- MultiAssayExperiment::MultiAssayExperiment(
+    experiments = list(
+      a = hermes::HermesDataFromMatrix(
+        matrix(1:4, 2, 2, dimnames = list(c("ENSG1", "ENSG2"), c("E", "F"))),
+        colData = data.frame(USUBJID = c("A", "B")) # USUBJID on experiment.
+      )
+    ),
+    colData = data.frame(
+      USUBJID = c("A", "B"),
+      row.names = c("E", "F") # USUBJID on sample map.
+    )
+  )
+  adtte <- data.frame(USUBJID = c("E", "F"))
+
+  expect_error(
+    h_km_mae_to_adtte(
+      adtte,
+      mae,
+      genes = hermes::gene_spec("ENSG1"),
+      experiment_name = "a"
+    ),
+    "Must be a subset of {'E','F'}, but has additional elements {'A','B'}",
+    fixed = TRUE
+  )
+})
+
+# adtteSpecInput ----
+
+test_that("adtteSpecInput creates expected HTML", {
+  expect_snapshot(adtteSpecInput(
+    "adtte",
+    label_paramcd = "Select right PARAMCD"
+  ))
+})
+
+test_that("{shinytest2} recording: adtteSpecServer module works as expected in the test app", {
+  skip_if_covr()
+  skip_if_too_deep(5)
+
+  app <- AppDriver$new(
+    app_dir = "adtteSpec",
+    name = "adtteSpecServer module works as expected in the test app"
+  )
+  ns <- module_ns_shiny2(app)
+  ns2 <- NS("teal-main_ui-filter_panel")
```

Comment:
can put this centrally since it does not depend on the module

Reply:
moved to  `helper-shinytest2.R`